api/openapi.yaml

```
@@ -1244,184 +1438,272 @@ components:
+          example: 01ARZ3NDEKTSV4RRFFQ69G5FAV
         metadata:
+          description: |
+            Additional metadata for the feature.
           type: object
           additionalProperties:
             type: string
-          example:
-            stripePaymentId: "pi_4OrAkhLvyihio9p51h9iiFnB"
-    LedgerAlreadyExistsProblem:
-      description: Ledger Exists
-      x-go-type-import:
-        path: github.com/openmeterio/openmeter/internal/credit
-      x-go-type: credit.LedgerAlreadyExistsProblemResponse
-      type: object
-      # TODO: until https://github.com/deepmap/oapi-codegen/pull/1610 is not merged, we
-      # need to copy the Problem schema here
-      #
-      # Afterwards the schema should be this
-      #allOf:
-      #- $ref: "#/components/schemas/Problem"
-      #- type: object
-      #  properties:
-      #    conflictingEntity:
-      #      $ref: "#/components/schemas/Ledger"
-      properties:
-        type:
-          type: string
-          format: uri
-          description: Type contains a URI that identifies the problem type.
-          example: "urn:problem-type:bad-request"
-        title:
-          type: string
-          description: A a short, human-readable summary of the problem type.
-          example: Bad Request
-        status:
-          type: integer
-          format: int32
-          description: The HTTP status code generated by the origin server for this occurrence of the problem.
-          minimum: 400
-          maximum: 599
-          example: 400
-        detail:
-          type: string
-          description: A human-readable explanation specific to this occurrence of the problem.
-          example: "body must be a JSON object"
-        instance:
-          type: string
-          format: uri
-          description: A URI reference that identifies the specific occurrence of the problem.
-          example: "urn:request:local/JMOlctsKV8-000001"
-        conflictingEntity:
-          $ref: "#/components/schemas/Ledger"
-      additionalProperties: true
-      required:
-        - type
-        - title
-        - status
-        - detail
-      example:
-        type: "about:blank"
-        title: "Conflict"
-        status: 409
-        detail: "ledger (default.01HXVNDJR532E8GTBVSC2XK5D4) already exitst for subject subject-1"
-        instance: "urn:request:local/JMOlctsKV8-000001"
-        conflictingEntity:
-          id: 01HXVNDJR532E8GTBVSC2XK5D4
-          subject: "subject-1"
-    Ledger:
-      x-go-type-import:
-        path: github.com/openmeterio/openmeter/internal/credit
-      x-go-type: credit.Ledger
+        usagePeriod:
+          $ref: "#/components/schemas/RecurringPeriod"
+    EntitlementSharedFields:
+      allOf:
+        - $ref: "#/components/schemas/SharedMetaFields"
+        - $ref: "#/components/schemas/EntitlementCreateSharedFields"
+        - type: object
+          required:
+            - subjectKey
+          properties:
+            subjectkey:
```

```suggestion
            subjectKey:
```

Given that https://github.com/openmeterio/openmeter/blob/0651acb69cb1be32ca44f2cf59d08fefcff72d74/internal/entitlement/entitlement.go#L14

Also we seem to miss [MeasureUsageFrom](https://github.com/openmeterio/openmeter/blob/0651acb69cb1be32ca44f2cf59d08fefcff72d74/internal/entitlement/entitlement.go#L13).

---

yes, not everything is or will be implemented in this PR, thanks for the typo